ansible/generate_playbook.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/sh
+
+ANSIBLE_OPTIONS=""
+RUN=0
+
+function die() {
+    echo "ERROR: $1" >&2
+    exit 1
+}
+
+function generate_header() {
+    cat >"$1"<<-END
+---
+- hosts: all
+  roles:
+END
+}
+
+function generate_roles() {
+    local playbook="$1"
+    local roles=""
+    for role in $(find /etc/ansible/roles/* roles/* -maxdepth 0 -type d); do
+        roles="$roles $(basename "$role")"
+    done
+
+    [ "$roles" = "" ] && die "No roles found"
+
+    for role in $(echo "$roles" | tr ' ' '\n' | sort | uniq); do
+        cat >> "$playbook" <<-END
+    - role: $role
+      when: '"$role" in foreman_roles'
```

Comment:
maybe `foreman_ansible_roles` would be better to avoid conflicts with other params in foreman?